File: algorithms/sorting/selection_sort.py
import random
import numpy as np

""" arr = []
for i in range(1000):
    arr.append(float('{:5.2f}'.format(random.random() * 100)))

"""

arr = np.arange(1000)

def selection_sort(arr):

    for j in range(len(arr)-1):
        min = arr[j]

        for k in range(j+1, len(arr)):
            if arr[k] < min:
              min = arr[k]
              min_index = k
              arr[j], arr[min_index] = min, arr[j]  
    
    return arr

print(selection_sort(arr))

File: algorithms/sorting/selection_sort.js
let arr = new Array();

for (let i = 0; i < 10; i++) {
  arr.push(Math.random().toFixed(4) * 100);
}




function selectionSort(arr) {
  
  for (let j = 0; j < arr.length-1; j++) {
    let min = arr[j];
    let min_index = j;
    

    for (let k = j+1; k < arr.length; k++) {
      if (arr[k] < min) {
        min = arr[k];
        min_index = k;
      }  
    }
    let aux = 0;
    aux = arr[j];
    arr[j] = min;
    arr[min_index] = aux;
    console.log(arr);
  }
  return arr;
}


console.log(arr);